File: tang-generator/src/main/java/com/tang/generator/core/autoconfigure/GeneratorProperties.java
package com.tang.generator.core.autoconfigure;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

/**
 * 代码生成配置
 *
 * @author Tang
 */
@Configuration
@ConfigurationProperties(GeneratorProperties.GENERATOR_PREFIX)
@PropertySource("classpath:generator.yml")
public class GeneratorProperties {

    public static final String GENERATOR_PREFIX = "generator";

    @Value("${author}")
    private String author;

    @Value("${package-name}")
    private String packageName;

    @Value("${remove-pre}")
    private String removePre;

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getPackageName() {
        return packageName;
    }

    public void setPackageName(String packageName) {
        this.packageName = packageName;
    }

    public String getRemovePre() {
        return removePre;
    }

    public void setRemovePre(String removePre) {
        this.removePre = removePre;
    }

}

File: tang-tool/tang-generator/src/main/kotlin/com/tang/generator/autoconfigure/GeneratorProperties.kt
package com.tang.generator.autoconfigure

import com.tang.commons.factory.YamlPropertyLoaderFactory
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.PropertySource

/**
 * 代码生成配置
 *
 * @author Tang
 */
@Configuration
@ConfigurationProperties(GeneratorProperties.GENERATOR_PREFIX)
@PropertySource(value = ["classpath:generator.yml"], factory = YamlPropertyLoaderFactory::class)
class GeneratorProperties {

    companion object {
        const val GENERATOR_PREFIX = "generator"
    }

    var author: String = "Tang"

    var packageName: String = "com.tang.system"

    var removePre: String = "t_,tb_"

}
